### PR TITLE
add streamlit-searchbox

### PR DESCRIPTION
**Pull request by ytausch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ytausch** [comment]
@conda-forge/help-python ready for review!

**Comment by jaimergp** [comment]
Let's make the tests pass first :P 

Also, this is adding a `tests` top-level package (see clobber warnings), so we should patch their pyproject file so `tests/` is excluded from the package.

**Comment by ytausch** [comment]
Hopefully, the recipe is good now. I had to upgrade the minimum streamlit versions because the really old ones have incorrect dependency constraints. As a sideline, I attempt to fix that here:

- https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/893

File: recipes/streamlit-searchbox/0001-remove-tests-from-distribution.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+diff --git a/setup.py b/setup.py
+index 7480aee..8ebf3c3 100644
+--- a/setup.py
++++ b/setup.py
+@@ -9,7 +9,7 @@ setuptools.setup(
+     + "and provides a list of suggestions based on a provided function",
+     long_description_content_type="text/plain",
+     url="https://github.com/m-wrzr/streamlit-searchbox",
+-    packages=setuptools.find_packages(),
++    packages=setuptools.find_packages(exclude=("tests",)),
```
Comment (ocefpaf) [inline recipes/streamlit-searchbox/0001-remove-tests-from-distribution.patch:27]: If you can, please report this upstream.

**Pull request merged**